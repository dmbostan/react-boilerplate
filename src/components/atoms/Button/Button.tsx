import {ButtonTypes} from "./Button.types.ts";

const Button = ({clickHandler, className, children}: ButtonTypes) => {
    return(
        <button onClick={clickHandler} className={className}>
            {children}
        </button>
    );
};

export default Button;
