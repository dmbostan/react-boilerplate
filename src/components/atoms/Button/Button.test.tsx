import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button component', () => {
    it('renders a button with the provided text and class name', () => {
        const clickHandler = vi.fn();
        const className = 'custom-button';
        const buttonText = 'Click me';

        const { getByText } = render(
            <Button clickHandler={clickHandler} className={className}>
                {buttonText}
            </Button>
        );

        const buttonElement = getByText(buttonText);

        expect(buttonElement).toBeInTheDocument();
        expect(buttonElement).toHaveClass(className);
    });

    it('calls the click handler when clicked', () => {
        const clickHandler = vi.fn();
        const className = 'custom-button';
        const buttonText = 'Click me';

        const { getByText } = render(
            <Button clickHandler={clickHandler} className={className}>
                {buttonText}
            </Button>
        );

        const buttonElement = getByText(buttonText);
        fireEvent.click(buttonElement);

        expect(clickHandler).toHaveBeenCalled();
    });

    it('matches snapshot', () => {
        const clickHandler = vi.fn();
        const className = 'custom-button';
        const buttonText = 'Click me';

        const { asFragment } = render(
            <Button clickHandler={clickHandler} className={className}>
                {buttonText}
            </Button>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
