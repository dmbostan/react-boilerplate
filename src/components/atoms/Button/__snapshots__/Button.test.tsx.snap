// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button component > should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
  >
    Click me
  </button>
</DocumentFragment>
`;
